Kernel/VirtIO/VirtIOConsole.cpp

```diff
@@ -91,12 +124,26 @@ void VirtIOConsole::handle_queue_update(u16 queue_index)
 
 bool VirtIOConsole::can_read(const FileDescription&, size_t) const
 {
-    return true;
+    return m_receive_buffer->used_bytes() > 0;
 }
 
-KResultOr<size_t> VirtIOConsole::read(FileDescription&, u64, [[maybe_unused]] UserOrKernelBuffer& data, size_t)
+KResultOr<size_t> VirtIOConsole::read(FileDescription& desc, u64, UserOrKernelBuffer& buffer, size_t size)
 {
-    return ENOTSUP;
+    if (!size)
+        return 0;
+
+    if (!can_read(desc, size))
+        return EAGAIN;
+
+    ScopedSpinLock ringbuffer_lock(m_receive_buffer->lock());
+    auto& queue = get_queue(RECEIVEQ);
+    ScopedSpinLock queue_lock(queue.lock());
+
+    auto bytes_copied = m_receive_buffer->copy_data_out(size, buffer);
+    m_receive_buffer->reclaim_space(m_receive_buffer->start_of_used(), bytes_copied.value());
+    evaluate_block_conditions();
```

Comment:
why are we evaluating block conditions? we just read form the receive buffer, so it has less available data, not more, so if someone was blocked he would still be blocked.